images/macos/provision/core/haskell.sh

```diff
@@ -1,5 +1,13 @@
 #!/bin/bash -e -o pipefail
 
+source ~/utils/utils.sh
+
+# https://github.com/actions/runner-images/issues/8738
+if is_BigSur; then
+    (brew list ghc > /dev/null 2>&1) && brew unlink ghc
+    (brew list cabal-install > /dev/null 2>&1) && brew unlink cabal-install
```

Comment:
I don't think you need `brew list` here. It doesn't change behavior at all. For example, about `ghc`:

- If you have `ghc` installed then both commands succeed and execution proceeds either with or without `brew list`
- If you don't have `ghc` installed then both commands fail and execution stops either with or without `brew list`